Makefile

```diff
@@ -1,17 +1,37 @@
 REPO := quay.io/boson/faas
 BIN  := faas
 
+WINDOWS=$(BIN)_windows_amd64.exe
+LINUX=$(BIN)_linux_amd64
+DARWIN=$(BIN)_darwin_amd64
+
 CODE := $(shell find . -name '*.go')
 DATE := $(shell date -u +"%Y%m%dT%H%M%SZ")
 HASH := $(shell git rev-parse --short HEAD 2>/dev/null)
 VTAG := $(shell git tag --points-at HEAD)
 VERS := $(shell [ -z $(VTAG) ] && echo 'tip' || echo $(VTAG) )
 
-all: $(BIN)
+all: $(WINDOWS) $(LINUX) $(DARWIN)
```

Comment:
This new flow always explicitly sets the OS and Architecture, which means someone checking out the repository from a different combination (such as on ARM)) can no longer run `make` as is instructed in the documentation.

I would perhaps consider leaving the current default make target in place, and instead creating a _new_ `make dist` target which has as its contents these three build commands.  Ideally, it would also create the directory `./dist`, and build the binaries into that directory, and remove the directory entirely on `make clean`.

This minor change would also allow for us to avoid having to update the `Dockerfile` and `.dockerignore`.

Reply:
That's a good idea, but since the current flow already left the faas binary on the main folder, can I consider this a future enhancement ?

Reply:
I'll leave the default as Linux (as before) but create a new target that is cross-platform.